src/lib/utility/Cluster.js

```diff
@@ -0,0 +1,64 @@
+export default class Cluster {
+    #items = [];
+
+    #group;
+
+    #startTime;
+
+    constructor(group) {
+      this.#group = group;
+    }
+
+    add(item) {
+      this.#items.push(item);
+    }
+
+    setStart(startTime) {
+      this.#startTime = startTime;
+    }
+
+    get length() {
+      return this.#items.length;
+    }
+
+    get startTime() {
+      return this.#startTime();
+    }
+
+    get itemsInCluster() {
+      return this.#items;
+    }
+
+    #clusterHasRequiredFields() {
+      if (!this.#items) {
+        throw new Error('Please add items to your cluster');
```

Comment:
We couldnt also maybe fail this case silently by just ignoring empty clusters vs throwing an error.

Reply:
By the time someone gets to this state, they have created a cluster with a group id and have tried to call .toObject() on it. If they forgot to add an item to the cluster, I would rather throw an error vs returning an empty object or undefined vs having wonder why they are missing a cluster somewhere on their screen.  Maybe they tried to add an item, but the item is null or undefined.